Predavanje_08/zakljucna_naloga.Rmd

```diff
@@ -0,0 +1,157 @@
+znacilke <- c( "WindSpeed", "Humidity", "Pressure", "Temp" )
+
+for (z in znacilke){
+  z2 <- paste0(z, c("9am", "3pm"))
+  data[,z] <- apply(data[, z2], 1, max)
+  data <- data[, - which(colnames(data) %in% z2)]
+}
+head(data)
+```
+
+6) Izpišite kateri dan je bil najbolj vetroven in hkrati tudi deževen.
+```{r echo = F}
+# Na grd nacin
+# tmp <- data[data$RainToday == 'Yes',]
+# tmp[which.max(data$WindSpeed[data$RainToday == 'Yes']), "Date"]
+
+# Lepsi nacin
+data %>% filter(RainToday=='Yes', WindSpeed == max(WindSpeed)) %>% select(Date)
+```
+
+7) Izpišite kolikokrat se je zgodilo, da sta bila dva dneva zapored deževna.
+```{r echo = F}
+nrow(data[data$RainToday == 'Yes' &data$RainTomorrow == 'Yes',])
+```
+
+8) Shranite tabelo v .xlsx format z imenom `clean_weatherAUS.xlsx`.
+```{r echo = F}
+write.xlsx(data, "data_raw/clean_weatherAUS.xlsx", overwrite = T)
+```
+
+## 2. Del: Vizualizacije
+
+1) Narišite raztrosni graf, ki prikazuje vlažnost v odvisnosti temperature. Točke obarvajte različno glede na to, če je tisti dan deževen ali ne.
+
+```{r echo = F, fig.width=4, fig.height= 3}
+ggplot(data, aes(x = Temp, y = Humidity, color = RainToday)) + geom_point()
+```
+
+
+2) Prikažite s črtnim diagramom vrednosti stolpcov `Weeks` in `Months` za obdobje med "2010-01-01" in "2011-04-01". Graf naj vsebuje obe črti istočasno.
+```{r echo = F}
+df <- data[(data$Date > as.Date("2010-01-01")) &( data$Date < as.Date("2011-04-01")), c("Date", "Weeks", "Months")]
+df <- pivot_longer(df, cols = c("Weeks", "Months"), names_to = "Data", values_to = "value")
+```
+```{r echo = F, fig.width=5, fig.height= 4}
+ggplot(df, aes(x = Date, y = value, color = Data)) + geom_line() + geom_point() + ylab("")
+```
+
+3) 
+  a. S stolpičnim diagramom prikažite povprečne dnevne padavine za letni kvartil.
+```{r echo = F}
+df2 <- data[, c("Date", "Rainfall")]
+df2$Quarter <- factor(quarter(df2$Date))
+```
+```{r echo = F, fig.width=4, fig.height= 3}
+ggplot(data = df2, aes(x = Quarter, y = Rainfall)) + geom_bar(stat = "summary", fun = mean) + ylab("Rainfall (mm)")
+```
+
+  b. (Težje) Stolpcom dodajte tudi `errorbar`, ki predstavlja standardno napako (sd(x)/sqrt(length(x))).
+```{r echo = F}
+se <- function(x) return (sd(x)/sqrt(length(x)))
+df3 <- data.frame( mean = tapply(df2$Rainfall, df2$Quarter, mean),
+                   sd = tapply(df2$Rainfall, df2$Quarter, se))
+
+df3$Quarter <- levels(df2$Quarter)
+```
+```{r echo = F, fig.width=4, fig.height= 3}
+ggplot(data = df3, aes(x = Quarter, y = mean, fill = Quarter)) + geom_bar(stat = "identity", width = 0.5) + geom_errorbar(aes(ymin = mean-sd, ymax = mean+sd), stat = "identity", width = 0.1, size = 1.1) + ylab("Rainfall (mm)")
+```
+  
+4. Za stolpce "Pressure", "Humidity" in "Temp" prikažite črtne diagrame s točkami za obdobje med "2012-01-01" in "2013-01-01". Če je bil naslednji dan deževen, točko obarvajte z drugo barvo. Iz preglednosti, vsako meritev prikažite na "pod-grafu" oziroma, če je pretežko, na ločenih grafih.
```

Comment:
- Kadar se imena nanašajo na stolpce data.frame bi predlagal, da uporabiš ``. Podobno kot pri P06, da smo konsistentni (čeprav nisem prepričan, da smo povsod to uporabili).
- Datume piši v konsistentnem formatu čez celotno nalogo.
- "Iz preglednosti, vsako meritev prikažite na "pod-grafu" oziroma, če je pretežko, na ločenih grafih." Tega stavka ne razumem. Mogoče brez "Iz preglednosti"?